NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -162,15 +163,56 @@ public StateQuery()
                                     }
                                 }
 #pragma warning disable CS0618 // Type or member is obsolete
-                                arenastate.OrderedArenaInfos.AddRange(arenaInfos.OrderByDescending(a => a.Score).ThenBy(a => a.CombatPoint));
+                                arenastate.OrderedArenaInfos.AddRange(arenaInfos.OrderByDescending(a => a.Score)
+                                    .ThenBy(a => a.CombatPoint));
 #pragma warning restore CS0618 // Type or member is obsolete
                             }
                         }
+
                         return arenastate;
                     }
 
                     return null;
                 });
+            Field<NonNullGraphType<ListGraphType<ArenaInformationType>>>(
+                name: "arenaInformation",
+                description: "List of arena information of requested arena and avatar list",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "championshipId",
+                        Description = "Championship ID to get arena information"
+                    },
+                    new QueryArgument<NonNullGraphType<IntGraphType>>
+                    {
+                        Name = "round",
+                        Description = "Round of championship to get arena information"
+                    },
+                    new QueryArgument<NonNullGraphType<ListGraphType<NonNullGraphType<AddressType>>>>
+                    {
+                        Name = "avatarAddresses",
+                        Description = "List of avatar address to get arena information"
+                    }
+                ),
+                resolve: context =>
+                {
+                    var championshipId = context.GetArgument<int>("championshipId");
+                    var round = context.GetArgument<int>("round");
+                    return context.GetArgument<List<Address>>("avatarAddresses").AsParallel().AsOrdered().Select(
+                        address =>
+                        {
+                            var infoAddr = ArenaInformation.DeriveAddress(address, championshipId, round);
+                            var scoreAddr = ArenaScore.DeriveAddress(address, championshipId, round);
+
+                            return (
+                                address,
+                                new ArenaInformation((List)context.Source.GetState(infoAddr)!),
+                                new ArenaScore((List)context.Source.GetState(scoreAddr)!)
```

Comment:
When there is not the `avatar`'s arena information, doesn't it return `null`? 🤔 I thought this question because this query's type is `List[Optional[ArenaInformation]]` (in Python expression)

Reply:
In fact, I ignored this nullable part intentionally. It's a kind of believing that who has created `BattleArena` action must be created `JoinArena` action before thus `ArenaInformation` and `ArenaScore` is not nullable.
As you pointed out, it is good to change return type from `<NonNullGraphType<ListGraphType<ArenaInformationType>>>` to `<NonNullGraphType<ListGraphType<NonNullGraphType<ArenaInformationType>>>>` to ensure everyone has information.